sql/clients_last_seen_v1.init.sql

```diff
@@ -0,0 +1,17 @@
+SELECT
+  @submission_date AS submission_date,
+  CURRENT_DATETIME() AS generated_time,
+  MAX(submission_date_s3) AS last_seen_date,
+  -- approximate LAST_VALUE(input).*
+  ARRAY_AGG(input
+    ORDER BY submission_date_s3
+    DESC LIMIT 1
+  )[OFFSET(0)].* EXCEPT (submission_date_s3)
```

Comment:
This is fascinating. I like this better than having to use a ROW_NUMBER window function and then select `n = 1`.

Reply:
we could alternately `ANY_VALUE(LAST_VALUE(input) OVER (ORDER BY submission_date_s3))`, but i don't know the performance implications of that

Reply:
> but i don't know the performance implications of that

I decided to check and it's not as simple as above, but using a window is so much faster it hurts